brightnessctl.c

```diff
@@ -601,12 +602,16 @@ bool ensure_dev_dir(struct device *dev) {
 	return ret;
 }
 
-char *_cat_with(char c, ...) {
+char *_cat_with(int c, ...) {
+	// We'd like c to be a char, but passing a char to va_start triggers
+	// undefined behavior. Take it as an int instead, and assert that it can
+	// fit in a char before using it as one.
+	assert(c >= CHAR_MIN && c <= CHAR_MAX);
 	size_t size = 32;
 	size_t length = 0;
 	char *buf = calloc(1, size + 1);
 	char *curr;
-	char split[2] = {c, '\0'};
+	char split[2] = {(char) c, '\0'};
```

Comment:
No need for explicit conversion here, the demotion is defined by the standard (If we are being pedantic)